db/migrate/20190102161442_add_taggings_for_all_taggable_tables.rb

```diff
@@ -0,0 +1,56 @@
+class AddTaggingsForAllTaggableTables < ActiveRecord::Migration[5.1]
+  def change
+    create_table "tags", id: :serial, force: :cascade do |t|
+      t.references :tenant, :type => :bigint, :index => false, :null => false, :foreign_key => {:on_delete => :cascade}
+
+      t.string "name", :null => false
+      t.text "description"
+      t.boolean "active", :default => true, :null => false
+
+      # TODO what is human readable name? Will we use it for searching? How will we search with translated field?
+      t.string "display_name"
+
+      # TODO the queries like "starts_with" won't be consistent across platform, if we'll model separately vs. encoded
+      # as namespace/name. Or we'll have to do query transformations, will we do that?
+      t.string "namespace"
+
+      # TODO how to model 'Whether this tag can accept zero, one, or multiple associated values', should be just
+      # 'multiple' boolean and allow it only if there are values to be associated?
+      t.boolean "multiple", :default => false, :null => false
```

Comment:
@Fryguy @agrare please advise. Are we going to add available tag values to it's own table (tag_values)? Or we'll encode the value into the name?

Also will we use the same approach for the mapped tags? So e.g. taking aws tag ```Name``` the tag_values would contain all unique names.